### ISSUE TITLE
Crash when trying to load a non-existent local resource

### ISSUE THREAD

**Issue by benedict-lim** [GRizzi91/bouquet#20]
Thank you for the prompt release of `1.0.2`. Much appreciated.

I ran into another problem when trying to test `state.error` by purposely opening a non-existent local resource.

```
E  Writing exception to parcel
java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaDocumentsProvider uri content://com.android.providers.media.documents/document/test from pid=14210, uid=10320 requires that you obtain access using ACTION_OPEN_DOCUMENT or related APIs
    at android.content.ContentProvider.enforceReadPermissionInner(ContentProvider.java:879)
    at android.content.ContentProvider$Transport.enforceReadPermission(ContentProvider.java:707)
    at android.content.ContentProvider$Transport.enforceFilePermission(ContentProvider.java:697)
    at android.content.ContentProvider$Transport.openTypedAssetFile(ContentProvider.java:556)
    at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:327)
    at android.os.Binder.execTransactInternal(Binder.java:1280)
    at android.os.Binder.execTransact(Binder.java:1244)
```

Would putting the file opening logic within a try-catch statement solve the problem?

https://github.com/GRizzi91/bouquet/blob/6f2f198c2d0a052ca4bb4000a8620603ec32b65c/bouquet/src/main/java/com/rizzi/bouquet/Bouquet.kt#L249-L259

**Comment by GRizzi91** [comment]
@benedict-lim Hi, sure i can add and exception management, i didn't read the documentation because i guess if `openFileDescritpor` didn't find a file it will just return null. All in all in that case the method will rise a `FileNotFoundException`, in your case it rise a `SecurityException: Permission Denial` so your problem is not the non-existence of the file but you are trying to access to a folder that you don't have permission to access.

**Comment by GRizzi91** [comment]
Fixed in version `1.1.1`

**Comment by benedict-lim** [comment]
@GRizzi91 
Sorry for the late response! Thank you for the fix. Much appreciated.
